lib/logstash/inputs/stdin.rb

```
@@ -31,16 +36,21 @@ def run(queue)
         end
       rescue IOError, EOFError, LogStash::ShutdownSignal
         # stdin closed or a requested shutdown
+        @stop_requested.make_true
         break
+      rescue => e
+        # ignore any exception in the shutdown process
+        break if @stop_requested.true?
+        raise(e)
```

Ok you are ignoring exception, its a short circuit.